packages/aws-cdk-lib/aws-dynamodb/lib/table-v2.ts

```diff
@@ -485,9 +525,15 @@ export class TableV2 extends TableBaseV2 {
       this.addKey(props.sortKey, RANGE_KEY_TYPE);
     }
 
-    this.billingMode = props.billing?.mode ?? BillingMode.PAY_PER_REQUEST;
-    this.readProvisioning = props.billing?._renderReadCapacity();
-    this.writeProvisioning = props.billing?._renderWriteCapacity();
+    if (props.billing?.mode === BillingMode.PAY_PER_REQUEST || props.billing?.mode === undefined) {
```

Comment:
for clarification, what we're trying to do here is check if customer has set billing mode, even if its undefined by they have the maxReadRequestUnits and maxWriteRequestUnits, billing mode should be set to `PAY_PER_REQUEST`.
Let me know if my understanding is not correct.

From the code it looks like, what will happen is even if you get undefined values from renderCapacity functions your billing mode will be set to `PAY_PER_REQUEST`. If this is not intended, instead of simply checking for undefined billing mode we should validate and throw an error that props.billing.mode should be set in order to use new props.